daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Util.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package value
+
+import com.daml.lf.transaction.TransactionVersion
+import com.daml.lf.value.Value._
+
+object Util {
+
+  // equivalent to serialization + unserialization.
```

Comment:
worth mentioning that it will never add types? so if you have something of an older version without types it won’t get normalized.

Reply:
- old (de)serialization processes accepts values without type information -- We just never used this possibility.
- the new serialization drops type information if present, while the new deserialization check it is not there.